ci/default.nix

```diff
@@ -21,9 +21,42 @@ let
     config = { };
     overlays = [ ];
   };
+
+  fmt =
+    let
+      treefmtNixSrc = fetchTarball {
+        # Master at 2025-02-12
+        url = "https://github.com/numtide/treefmt-nix/archive/4f09b473c936d41582dd744e19f34ec27592c5fd.tar.gz";
+        sha256 = "051vh6raskrxw5k6jncm8zbk9fhbzgm1gxpq9gm5xw1b6wgbgcna";
+      };
```

Comment:
Now we have `treefmt.withConfig` and/or `nixfmt-tree` I don't think we need this dependency on treefmt-nix.

Or maybe `config.build.devShell` is doing something important, that adding `treefmt.withConfig {}` to a shell wouldn't do?

(This is minor enough to delay for another PR, especially given how late my comment is)